src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -242,45 +242,94 @@ Namespace Microsoft.CodeAnalysis.Semantics
                             isImplicit = True
                         End If
                         initializer = OperationFactory.CreateVariableInitializer(initializerSyntax, initializerValue, _semanticModel, isImplicit)
                     End If
-                    builder.Add(OperationFactory.CreateVariableDeclaration(declaration.LocalSymbol, initializer, _semanticModel, declaration.Syntax))
-                ElseIf base.Kind = BoundKind.AsNewLocalDeclarations Then
-                    Dim asNewDeclarations = DirectCast(base, BoundAsNewLocalDeclarations)
-                    Dim localSymbols = asNewDeclarations.LocalDeclarations.SelectAsArray(Of ILocalSymbol)(Function(declaration) declaration.LocalSymbol)
+                ElseIf first.Kind = BoundKind.AsNewLocalDeclarations Then
+                    Dim asNewDeclarations = DirectCast(first, BoundAsNewLocalDeclarations)
+                    singleDeclarations = asNewDeclarations.LocalDeclarations.SelectAsArray(AddressOf GetSingleVariableDeclaration)
                     Dim initializerSyntax As AsClauseSyntax = DirectCast(asNewDeclarations.Syntax, VariableDeclaratorSyntax).AsClause
                     Dim initializerValue As IOperation = Create(asNewDeclarations.Initializer)
-                    Dim initializer As IVariableInitializer = OperationFactory.CreateVariableInitializer(initializerSyntax, initializerValue, _semanticModel, isImplicit:=False)
-                    builder.Add(OperationFactory.CreateVariableDeclaration(localSymbols, initializer, _semanticModel, asNewDeclarations.Syntax))
+                    initializer = OperationFactory.CreateVariableInitializer(initializerSyntax, initializerValue, _semanticModel, isImplicit:=False)
                 End If
+
+                builder.Add(New MultiVariableDeclaration(singleDeclarations,
+                                                         initializer,
+                                                         _semanticModel,
+                                                         declarationGroup.Key,
+                                                         type:=Nothing,
+                                                         constantValue:=Nothing,
+                                                         isImplicit:=False))
             Next
 
             Return builder.ToImmutableAndFree()
         End Function
 
-        Private Function GetUsingStatementDeclaration(resourceList As ImmutableArray(Of BoundLocalDeclarationBase), syntax As SyntaxNode) As IVariableDeclarationStatement
+        Private Function GetSingleVariableDeclaration(boundLocalDeclaration As BoundLocalDeclaration) As ISingleVariableDeclaration
+            Dim initializer As Lazy(Of IVariableInitializer) = New Lazy(Of IVariableInitializer)(
+                Function()
+                    If boundLocalDeclaration.ArrayCreationOpt IsNot Nothing Then
+                        Dim arrayBoundsSyntax = boundLocalDeclaration.Syntax
```

Comment:
>arrayBoundsSyntax [](start = 28, length = 17)

Is this really the array bounds syntax?

Reply:
It was at one point, until I realized that was wrong.

---
In reply to: [146976094](https://github.com/dotnet/roslyn/pull/22842#discussion_r146976094) [](ancestors = 146976094)